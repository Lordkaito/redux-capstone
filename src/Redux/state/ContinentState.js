import Axios from 'axios';

const URL = 'https://api.covid19tracking.narrativa.com/api';
const GET_ALL_COUNTRIES = 'rrcapstone/state/GET_ALL_COUNTRIES';

export const fetchDataForAllCountries = () => async (dispatch) => {
  const objStore = {};
  const today = new Date();
  const date = `${today.getFullYear()}-${today.getMonth() + 1}-${today.getDate() - 1}`;
  const response = await Axios.get(`${URL}/${date}`);
  const data = response.data.dates[date].countries;
  const countries = Object.keys(data);
  countries.forEach((country) => {
    objStore[country] = data[country];
  });
  dispatch({
    type: GET_ALL_COUNTRIES,
    payload: objStore,
  });
};

const initialState = [];

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_ALL_COUNTRIES:
      return action.payload;
    default:
      return state;
  }
};

export default reducer;