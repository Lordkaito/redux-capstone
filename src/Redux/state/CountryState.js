import Axios from 'axios';

const URL = 'https://api.covid19tracking.narrativa.com/api';
const GET_COUNTRY = 'rrcapstone/state/GET_COUNTRY';

export const fetchDataForCountry = (country) => async (dispatch) => {
  // fetch data from the API
  const myCountry = { ...country };
  dispatch({
    type: GET_COUNTRY,
    payload: myCountry,
  });
};

const initialState = [];

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_COUNTRY:
      return action.payload;
    default:
      return state;
  }
};

export default reducer;