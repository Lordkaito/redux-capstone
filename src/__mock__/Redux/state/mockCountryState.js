import Axios from 'axios';

const URL = 'https://api.covid19tracking.narrativa.com/api';
const GET_COUNTRY = 'rrcapstone/state/GET_COUNTRY';

export const fetchDataForCountry = (country) => {
  // fetch data from the API
  const myCountry = { ...country };
  return ({
    type: GET_COUNTRY,
    payload: myCountry,
  });
};

const initialState = {
  date: '2021-11-18',
  id: 'albania',
  links: [
    {
      href: '/api/2021-11-18/country/albania',
      rel: 'self',
      type: 'GET',
    },
  ],
  name: 'Albania',
  name_es: 'Albania',
  name_it: 'Albania',
  regions: [],
  source: 'John Hopkins University',
  today_confirmed: 195021,
  today_deaths: 3022,
  today_new_confirmed: 549,
  today_new_deaths: 8,
  today_new_open_cases: 541,
  today_new_recovered: 0,
  today_open_cases: 61656,
  today_recovered: 130343,
  today_vs_yesterday_confirmed: 0.0028230285079600126,
  today_vs_yesterday_deaths: 0.002654280026542777,
  today_vs_yesterday_open_cases: 0.008852163953202874,
  today_vs_yesterday_recovered: 0,
  yesterday_confirmed: 194472,
  yesterday_deaths: 3014,
  yesterday_open_cases: 61115,
  yesterday_recovered: 130343,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_COUNTRY:
      return action.payload;
    default:
      return state;
  }
};

export default reducer;