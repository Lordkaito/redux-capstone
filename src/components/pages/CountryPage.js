import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import logo from '../images/map.png';
import '../styles/Country.css';

const CountryPage = () => {
  const myState = useSelector((state) => state.CountryReducer);
  return (
    <div className="country-container">
      <div className="map">
        <img src={logo} alt="map" className="map-icon" />
      </div>
      <div>
        <h1 data-testid="country-name-test"
          className="country-name">{myState.name}</h1>
      </div>
      <div className="country-info">
        <p>Date: {myState.date}</p>
        <p data-testid="source-test"
        >Source: {myState.source}</p>
        <p>Total confirmed: {myState.today_confirmed}</p>
        <p>New confirmed cases: {myState.today_new_confirmed}</p>
        <p>Total deaths: {myState.today_deaths}</p>
        <p>New deaths: {myState.today_new_deaths}</p>
        <p>New open cases: {myState.today_new_open_cases}</p>
        <p data-testid="total-test">Total recovered: {myState.today_recovered}</p>
      </div>
    </div>
  );
};

export default CountryPage;