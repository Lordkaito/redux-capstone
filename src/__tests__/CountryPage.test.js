import '@testing-library/jest-dom/extend-expect';
import React, { useEffect } from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import { Provider, useDispatch } from 'react-redux';
import { act } from 'react-dom/test-utils';
import CountryPage from '../components/pages/CountryPage';
import store from '../__mock__/Redux/mockConfigureStore';
import { fetchDataForCountry } from '../__mock__/Redux/state/mockCountryState';

describe('Main component rendering', () => {
  let CountryPageApp;
  act(() => {
    CountryPageApp = render(
      <React.StrictMode>
        <Router>
          <Provider store={store}>
            <CountryPage />
          </Provider>
        </Router>
      </React.StrictMode>,
    );
  });
  afterEach(() => {
    cleanup();
  });

  test('Country page test', () => {
    const nameElement = screen.getByTestId('country-name-test');
    expect(nameElement).toBeInTheDocument();
    expect(nameElement.innerHTML).toEqual('Albania');
    const sourceTest = screen.getByTestId('source-test');
    expect(sourceTest).toBeInTheDocument();
    expect(sourceTest.innerHTML).toContain('John Hopkins University');
    const totalTest = screen.getByTestId('total-test');
    expect(totalTest).toBeInTheDocument();
    expect(totalTest.innerHTML).toContain('130343');
  });

  test('snapshot of Main page with mocked store page', () => {
    expect(CountryPage).toMatchSnapshot();
  });
});