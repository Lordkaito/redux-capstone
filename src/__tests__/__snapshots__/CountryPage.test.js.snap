// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component rendering snapshot of Main page with mocked store page 1`] = `[Function]`;
